### PR TITLE
Render required permissions

### LINKED ISSUE #416: Fix UI for rules with required review by permissions
When reviews are required by permissions the UI does not display that fact:
Screen Shot 2022-05-03 at 11 56 25 AM

Rule:
```
  - name: one admin has approved (PR contributors not allowed)
    options:
      allow_contributor: false
    requires:
      count: 1
      permissions: ["admin", "maintain"]
```

### PR DESCRIPTION
**Pull request by ylee088**
Fixes #416.

### CODE REVIEW

File: server/handler/frontend.go
Comment (bluekeyes) [inline server/handler/frontend.go]: I think you'll want to use the new `GetPermissions` function I added in #426 (once it merges) here to make sure we're including all permissions and there are no duplicates.

Comment (ylee088) [reply server/handler/frontend.go]: Fixed, thank you!

Comment (ylee088) [reply server/handler/frontend.go]: Oh I totally missed the "once it merges" part. Failing now, but will wait until #426 is merged.

File: server/handler/frontend.go
Comment (bluekeyes) [inline server/handler/frontend.go]: When allocating a slice upfront and then filling it, you have two options

1. Use `make([]string, 0, length)`, then use `append` in the loop
2. Use `make([]string, length)`, then set elements by index in the loop

As written, this will create a slice with `len(result.Requires.Permissions)` empty strings, then append the new string on to the end of that.

I think you'll also want to initialize the slice using the length of `result.Requires.GetPermissions()` instead of `result.Requires.Permissions`.

All that said, another option is to remove this function and turn the `pull.Permission` types in to strings in the template, using the `printf` template function.

File: server/templates/details.html.tmpl
Comment (bluekeyes) [inline server/templates/details.html.tmpl]: Unlike the user/team/org section, `getPermissions` returns a `[]string` instead of a `map[string][]string`. I think you'll want to use a single `ul` rendered with one `range` expression for this, instead of the description list pattern.

**Comment by bluekeyes** [comment]
Also, looks like `server/.DS_Store` was added incorrectly - can you remove that file?

File: server/handler/frontend.go
Comment (ylee088) [reply server/handler/frontend.go]: Fixed, using option 1.

File: server/templates/details.html.tmpl
Comment (ylee088) [reply server/templates/details.html.tmpl]: Fixed, copy-pasta laziness :p

File: server/handler/frontend.go
Comment (bluekeyes) [inline server/handler/frontend.go]: I think this should check the result of `requires.GetPermissions()` to make sure it accounts for the legacy properties.

File: server/handler/frontend.go
Comment (bluekeyes) [inline server/handler/frontend.go]:
```suggestion
	perms := result.Requires.GetPermissions()
	permStrings := make([]string, 0, len(perms))
	for _, perm := range perms {
		permStrings = append(permStrings, perm.String())
	}
```

File: server/templates/details.html.tmpl
Comment (bluekeyes) [inline server/templates/details.html.tmpl]:
```suggestion
  <ul class="list-disc list-outside pl-6 py-2">
    {{range getPermissions .}}<li class="font-mono text-sm-mono">{{.}}</li>{{end}}
  </ul>
```

File: server/handler/frontend.go
Comment (ylee088) [reply server/handler/frontend.go]: Sorry, keep missing these!

**Review by bluekeyes — APPROVED** [review]

**Pull request merged**
